.github/workflows/unit_tests.yml

```
@@ -0,0 +1,38 @@
+name: unit_tests
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    branches:
+      - trunk
+
+jobs:
+  unit_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package:
+          - at_secondary_proxy
+        dart:
+          - stable
+          - beta
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+
+      - name: Set up Dart SDK
+        uses: dart-lang/setup-dart@e51d8e571e22473a2ddebf0ef8a2123f0ab2c02c
```

There should be a version comment after the hash:

```suggestion
        uses: dart-lang/setup-dart@e51d8e571e22473a2ddebf0ef8a2123f0ab2c02c # v1.7.1
```